The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10331763990/job/28602266794) [(plain)](https://github.com/rust-lang/rust/commit/d340d51d9663cac59762b06e52cfe982f71877c6/checks/28602266794/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 2.676 Building wheels for collected packages: reuse
#16 2.678   Building wheel for reuse (pyproject.toml): started
#16 2.925   Building wheel for reuse (pyproject.toml): finished with status 'done'
#16 2.927   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#16 2.927   Stored in directory: /tmp/pip-ephem-wheel-cache-biav9pk1/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#16 2.930 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#16 3.325 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#16 3.326 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#16 DONE 3.4s
---
    Checking powerfmt v0.2.0
    Checking miniz_oxide v0.7.3
    Checking deranged v0.3.11
   Compiling proc-macro-test v0.0.0 (/checkout/src/tools/rust-analyzer/crates/proc-macro-srv/proc-macro-test)
error[E0004]: non-exhaustive patterns: `&Constructor::F16Range(_, _, _)` and `&Constructor::F128Range(_, _, _)` not covered
    |
    |
374 |         let tys: SmallVec<[_; 2]> = match ctor {
    |                                           ^^^^ patterns `&Constructor::F16Range(_, _, _)` and `&Constructor::F128Range(_, _, _)` not covered
    |
note: `Constructor<MatchCheckCtx<'_>>` defined here
    |
    |
677 | pub enum Constructor<Cx: PatCx> {
...
...
693 |     F16Range(IeeeFloat<HalfS>, IeeeFloat<HalfS>, RangeEnd),
...
...
696 |     F128Range(IeeeFloat<QuadS>, IeeeFloat<QuadS>, RangeEnd),
    |     --------- not covered
    = note: the matched value is of type `&Constructor<MatchCheckCtx<'_>>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern, a match arm with multiple or-patterns as shown, or multiple match arms
429 ~             },
429 ~             },
430 +             &Constructor::F16Range(_, _, _) | &Constructor::F128Range(_, _, _) => todo!()

    Checking object v0.33.0
    Checking crc32fast v1.4.2
    Checking libloading v0.8.3
    Checking libloading v0.8.3
    Checking num-conv v0.1.0
    Checking unicode-ident v1.0.12
    Checking time-core v0.1.2
    Checking memmap2 v0.5.10
    Checking time v0.3.36
    Checking flate2 v1.0.30
    Checking byteorder v1.5.0
error[E0004]: non-exhaustive patterns: `&Constructor::F16Range(_, _, _)` and `&Constructor::F128Range(_, _, _)` not covered
    |
263 |         let kind = match pat.ctor() {
263 |         let kind = match pat.ctor() {
    |                          ^^^^^^^^^^ patterns `&Constructor::F16Range(_, _, _)` and `&Constructor::F128Range(_, _, _)` not covered
    |
note: `Constructor<MatchCheckCtx<'_>>` defined here
    |
    |
677 | pub enum Constructor<Cx: PatCx> {
...
...
693 |     F16Range(IeeeFloat<HalfS>, IeeeFloat<HalfS>, RangeEnd),
...
...
696 |     F128Range(IeeeFloat<QuadS>, IeeeFloat<QuadS>, RangeEnd),
    |     --------- not covered
    = note: the matched value is of type `&Constructor<MatchCheckCtx<'_>>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern, a match arm with multiple or-patterns as shown, or multiple match arms
313 ~             },
313 ~             },
314 +             &Constructor::F16Range(_, _, _) | &Constructor::F128Range(_, _, _) => todo!()


error[E0004]: non-exhaustive patterns: `&Constructor::F16Range(_, _, _)` and `&Constructor::F128Range(_, _, _)` not covered
    |
339 |         match ctor {
339 |         match ctor {
    |               ^^^^ patterns `&Constructor::F16Range(_, _, _)` and `&Constructor::F128Range(_, _, _)` not covered
    |
note: `Constructor<MatchCheckCtx<'_>>` defined here
    |
    |
677 | pub enum Constructor<Cx: PatCx> {
...
...
693 |     F16Range(IeeeFloat<HalfS>, IeeeFloat<HalfS>, RangeEnd),
...
...
696 |     F128Range(IeeeFloat<QuadS>, IeeeFloat<QuadS>, RangeEnd),
    |     --------- not covered
    = note: the matched value is of type `&Constructor<MatchCheckCtx<'_>>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern, a match arm with multiple or-patterns as shown, or multiple match arms
364 ~             },
364 ~             },
365 +             &Constructor::F16Range(_, _, _) | &Constructor::F128Range(_, _, _) => todo!()

   Compiling proc-macro-srv-cli v0.0.0 (/checkout/src/tools/rust-analyzer/crates/proc-macro-srv-cli)
    Checking zip v0.6.6
    Checking directories v5.0.1
```

</details>

--- a/src/tools/rust-analyzer/crates/hir-ty/src/lib.rs
+++ b/src/tools/rust-analyzer/crates/hir-ty/src/lib.rs
@@ -15,8 +15,10 @@ extern crate rustc_abi;
 #[cfg(not(feature = "in-rust-tree"))]
 extern crate ra_ap_rustc_abi as rustc_abi;
 
-// Use the crates.io version unconditionally until the API settles enough that we can switch to
-// using the in-tree one.
+#[cfg(feature = "in-rust-tree")]
+extern crate rustc_pattern_analysis;
+
+#[cfg(not(feature = "in-rust-tree"))]
 extern crate ra_ap_rustc_pattern_analysis as rustc_pattern_analysis;
 
 mod builder;

